src/lib/components/commons/FilterPanel/FilterPanel.tsx

```diff
@@ -0,0 +1,132 @@
+import { useState } from 'react';
+import { PROJECT_DISPLAY_TEXTS, ProjectFields } from '@/lib/consts/projects';
+import { useForm, FormProvider } from 'react-hook-form';
+import { ProjectFormValues } from '../../ProjectForm';
+import { projectFormSchema } from '../../ProjectForm/ProjectForm.consts';
+import { PROJECT_FORM_DEFAULT_VALUES } from '../../ProjectForm/ProjectForm.consts';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { SvgIcon } from '../../icons/SvgIcon';
+import {
+  StyledFilterPanel,
+  StyledFilterControlDiv,
+  StyledFilterButton,
+} from './FilterPanel.styled';
+import { StatusState } from './FilterPanel.types';
+import { DateInput } from '../Input/inputs/DateInput';
+
+export const FilterPanel = () => {
+  const [showFilters, setShowFilters] = useState(false);
+  const [status, setStatus] = useState<StatusState>('cancel');
+
+  const form = useForm<ProjectFormValues>({
+    resolver: zodResolver(projectFormSchema),
+    defaultValues: PROJECT_FORM_DEFAULT_VALUES,
+    mode: 'onSubmit',
+  });
+
+  return (
+    <>
+      <SvgIcon
+        size='S'
+        onClick={() => {
+          setShowFilters(!showFilters);
+        }}
+      >
+        <use
+          href={`/img/icons.svg#icon-filter-${showFilters ? 'open' : 'close'}`}
+        ></use>
+      </SvgIcon>
+
+      <FormProvider {...form}>
+        {showFilters && (
+          <StyledFilterPanel>
+            <StyledFilterControlDiv>
+              <p style={{ flexBasis: '100%' }}>
+                {PROJECT_DISPLAY_TEXTS.he.fields.status}
+              </p>
+              <StyledFilterButton
+                isButtonGroup={true}
+                variant={
+                  status === 'cancel'
+                    ? 'secondary'
+                    : status
+                    ? 'primary'
+                    : 'secondary'
+                }
+                onClick={() =>
+                  setStatus(!status ? true : status === true ? 'cancel' : true)
```

Comment:
you can create toggle function for this (if you would change it to boolean):
```const toggle = setStatus(prevStatus=>!prevStatus)```
than you can use it all over

Reply:
and in general , changing state the depend on the prev state , must be with this callback version of setState action